### PR TITLE
feat: Include few more pre-commit checks

### PR DESCRIPTION
**Pull request by armenzg**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,10 +1,47 @@
...
+  - repo: https://github.com/asottile/reorder_python_imports
+    rev: v3.1.0
+    hooks:
+      - id: reorder-python-imports
+        args: [--py38-plus, --add-import, "from __future__ import annotations"]
+  - repo: https://github.com/asottile/add-trailing-comma
+    rev: v2.2.3
+    hooks:
+      - id: add-trailing-comma
+        args: [--py36-plus]
```
Comment (asottile-sentry) [inline .pre-commit-config.yaml:30]: if we're using black we won't need this one

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,10 +1,47 @@
...
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v2.32.1
+    hooks:
+      - id: pyupgrade
+        args: [--py38-plus]
+  - repo: https://github.com/pre-commit/mirrors-autopep8
+    rev: v1.6.0
+    hooks:
+      - id: autopep8
```
Comment (asottile-sentry) [inline .pre-commit-config.yaml:39]: nor this one

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -1,10 +1,47 @@
...
+  - repo: https://github.com/PyCQA/flake8
+    rev: 4.0.1
+    hooks:
+      - id: flake8
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.960
+    hooks:
+      - id: mypy
```
Comment (asottile-sentry) [inline .pre-commit-config.yaml:47]: this probably needs some `additional_dependencies` to satisfy the stubs

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -88,7 +88,7 @@ jobs:
 
       - uses: actions/setup-python@v3
         with:
-          python-version: "3.8"
+          python-version: "3.10"
```
Comment (armenzg) [inline .github/workflows/ci.yml:91]: To make it match what we use in production.

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -1,10 +1,37 @@
...
     hooks:
       - id: check-github-workflows
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.2.0
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: check-yaml
+      - id: debug-statements
+      # - id: name-tests-test
```
Comment (armenzg) [inline .pre-commit-config.yaml:18]: I will defer this to later to make the diff easier to read.

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -1,10 +1,37 @@
...
+      - id: pyupgrade
+        args: [--py38-plus]
+  # - repo: https://github.com/PyCQA/flake8
+  #   rev: 4.0.1
+  #   hooks:
+  #     - id: flake8
+  # - repo: https://github.com/pre-commit/mirrors-mypy
+  #   rev: v0.960
+  #   hooks:
+  #     - id: mypy
```
Comment (armenzg) [inline .pre-commit-config.yaml:37]: Ignoring these two for now.

File: cli.py [hunk 7]
```diff
@@ -1,12 +1,14 @@
+from __future__ import annotations
```
Comment (armenzg) [inline cli.py:1]: The remaining changes are producing when running `pre-commit run --all-files`.

File: .pre-commit-config.yaml (hunk 3 above)
Comment (armenzg) [reply .pre-commit-config.yaml]: I will look into this in the future.

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**